pkg/controller/core/core.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package core
+
+import (
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/kueue/pkg/cache"
+	"sigs.k8s.io/kueue/pkg/queue"
+)
+
+// SetupControllers sets up the core controllers. It returns the name of the
+// controller that failed to create and an error, if any.
+func SetupControllers(mgr ctrl.Manager, qManager *queue.Manager, cc *cache.Cache) (string, error) {
```

Comment:
Now that we are doing this, what about making them share a recorder 
`mgr.GetEventRecorderFor())` 
would make it easier to set events

Reply:
Only the job controller and the scheduler have event recorders, and they are outside of this package :)

Although we could have another package that instantiates the queue manager, the cache and the scheduler.

I think we should always keep the job controller separate.

Reply:
yup, I am wondering if moving forward it will be of use to have a recorder in the core controllers as well, we might want to set events into the CRD, no?